lib/uikit/buttons/scaled_gesture_detector.dart

```diff
@@ -0,0 +1,83 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/uikit/buttons/animated_pressed_scale.dart';
+import 'package:flutter_template/uikit/others/conditional_wrapper.dart';
+
+/// {@template scaled_gesture_detector.class}
+/// GestureDetector with click animation.
+/// {@endtemplate}
+class ScaledGestureDetector extends StatefulWidget {
+  /// The callback of pressing the button.
+  final VoidCallback? onTap;
+
+  /// @nodoc
+  final VoidCallback? onTapUp;
+
+  /// @nodoc
+  final VoidCallback? onTapDown;
+
+  /// @nodoc
+  final VoidCallback? onTapCancel;
+
+  /// The element that will shrink when clicked.
+  final Widget child;
+
+  /// Callback when the click status changes.
+  final ValueChanged<bool>? isOnPressChanged;
```

Comment:
Why do we need this callback? I think it is redundant as the scale logic is handled by the widget itself.